### PR TITLE
More zero throttle and relax factoring

### PR DESCRIPTION
**Pull request by peterbarker**
This is based on #5227 - it should not be merged before that one.

The patches here change behaviour for helicopters.  It appears that these changes would be for the better!

### CODE REVIEW

**Comment by OXINARF** [comment]
@peterbarker Rebase please.

**Comment by peterbarker** [comment]
On Wed, 27 Dec 2017, Francisco Ferreira wrote:

> @peterbarker Rebase please.

Done.

Should I split this up again?

The acro and stabilize patches should be 
able to go straight in as those aren't called for heli.

Similarly for autotune - mode is disabled in config.h for heli

The payload-place part looks to be identical to the 
zero_throttle_and_relax() function - same with the mode_guided patch.

Doing this split/separate PR would simplify this one.

**Comment by OXINARF** [comment]
Yep, splitting is a good idea. The ones I can't approve are:

- Drift
- Land
- PosHold

I'm not completely sure about Autotune too - it's disabled by default in Heli, but the macro looks like just a feature removed. Does it not work in Heli at all? If it doesn't, then we should probably do like Throw and just don't allow it to ever enter in Heli builds.

File: ArduCopter/mode_throw.cpp [hunk 1]
```diff
@@ -258,7 +258,7 @@ bool Copter::ModeThrow::throw_attitude_good()
 
 bool Copter::ModeThrow::throw_height_good()
 {
-    // Check that we are no more than 0.5m below the demanded height
+    // Check that we are with 0.5m of the demanded height
```
Comment (OXINARF) [inline ArduCopter/mode_throw.cpp]: Within?

**Comment by peterbarker** [comment]
@OXINARF split and comment fixed

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**